src/IssueViz.Security/Taint/TaintStore.cs

```diff
@@ -62,17 +54,10 @@ public void Set(IEnumerable<IAnalysisIssueVisualization> issueVisualizations)
                 throw new ArgumentNullException(nameof(issueVisualizations));
             }
 
-            taintVulnerabilities.Clear();
-
-            foreach (var issueViz in issueVisualizations)
-            {
-                taintVulnerabilities.Add(issueViz);
-            }
-        }
+            var oldIssues = taintVulnerabilities;
+            taintVulnerabilities = issueVisualizations;
```

Comment:
It would be safer to store a copy of the enumerable so this class won't be impacted by changes to the caller (e.g. if it's passed a non-reified enumerator).

Reply:
The class is still storing the original enumerable that was passed in, so it could be mutated externally.

Reply:
Oh that's what you meant; my bad. I thought you meant the passing to the event args.